connector/src/main/java/org/geysermc/connector/utils/AdvancementsUtils.java

```diff
@@ -0,0 +1,163 @@
+import org.geysermc.connector.network.translators.world.block.BlockTranslator;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class AdvancementsUtils {
+
+    // Used in UpstreamPacketHandler.java
+    public static final int ADVANCEMENTS_MENU_FORM_ID = 1341;
+    public static final int ADVANCEMENTS_LIST_FORM_ID = 1342;
+
+    /**
+     * Build a form for the given session with all advancement categories
+     *
+     * @param session The session to build the form for
+     */
+    public static SimpleFormWindow buildMenuForm(GeyserSession session) {
+        // Cache the language for cleaner access
+        String language = session.getClientData().getLanguageCode();
+        // Created menu window for advancement categories
+        SimpleFormWindow window = new SimpleFormWindow(LocaleUtils.getLocaleString("gui.advancements", language), "");
+
+        window.getButtons().add(new FormButton(LocaleUtils.getLocaleString("advancements.story.root.title", language) + " - " + LocaleUtils.getLocaleString("advancements.story.root.description", language)));
+        window.getButtons().add(new FormButton(LocaleUtils.getLocaleString("advancements.nether.root.title", language) + " - " + LocaleUtils.getLocaleString("advancements.nether.root.description", language)));
+        window.getButtons().add(new FormButton(LocaleUtils.getLocaleString("advancements.end.root.title", language) + " - " + LocaleUtils.getLocaleString("advancements.end.root.description", language)));
+        window.getButtons().add(new FormButton(LocaleUtils.getLocaleString("advancements.adventure.root.title", language) + " - " + LocaleUtils.getLocaleString("advancements.adventure.root.description", language)));
+        window.getButtons().add(new FormButton(LocaleUtils.getLocaleString("advancements.husbandry.root.title", language) + " - " + LocaleUtils.getLocaleString("advancements.husbandry.root.description", language)));
+
+        return window;
+    }
+
+    /**
+     * Handle the menu form response
+     *
+     * @param session  The session that sent the response
+     * @param response The response string to parse
+     * @return True if the form was parsed correctly, false if not
+     */
+    public static boolean handleMenuForm(GeyserSession session, String response) {
+        SimpleFormWindow menuForm = (SimpleFormWindow) session.getWindowCache().getWindows().get(ADVANCEMENTS_MENU_FORM_ID);
+        menuForm.setResponse(response);
+        SimpleFormResponse formResponse = (SimpleFormResponse) menuForm.getResponse();
+
+        // Cache the language for cleaner access
+        String language = session.getClientData().getLanguageCode();
+        StringBuilder content = new StringBuilder();
+        String title = null;
+        if (formResponse != null && formResponse.getClickedButton() != null) {
+            content.append(LanguageUtils.getPlayerLocaleString("geyser.advancements.earned", language) + ":\n");
```

Comment:
Where is the PR for this string?

Reply:
Check languages on EN-US

Reply:
https://github.com/GeyserMC/languages/pull/52